code/modules/hydroponics/trays/tray.dm

```diff
@@ -574,6 +574,29 @@
 		anchored = !anchored
 		to_chat(user, "<span class='filter_notice'>You [anchored ? "wrench" : "unwrench"] \the [src].</span>")
 
+	else if(istype(O, /obj/item/shovel/spade))
+		if(!weedlevel && !seed) // Don't do anything if the tray is truly empty
+			to_chat(user, SPAN_NOTICE("[src] doesn't have any plants or weeds!"))
+			return
+		user.visible_message(SPAN_NOTICE("[user] starts digging out [src]'s plants..."))
+		to_chat(user, SPAN_NOTICE("You start digging out [src]'s plants..."))
+		playsound(src, O.usesound, 50, 1)
+		if(do_after(user, 40 * O.toolspeed) || (!seed && !weedlevel))
```

Comment:
This is already checked above. Putting it here will be at the start of the do_after afaik, not the end.

Reply:
Having a little trouble understanding what you mean, but I hope I'm interpreting this correctly and updated it

Reply:
This specific condition is checked earlier in the function, so checking it here is redundant.
If it's meant to be checked _after_ the do_after() finishes, then I believe you need to put it inside of the scope of the if statement containing the do_after().
So yes, that is a way to resolve it, if you didn't intend to check that condition after the do_after().